src/Ursa/Controls/NumericUpDown/NumericUpDownBase.cs

```diff
@@ -343,18 +343,38 @@ protected static string TrimString(string? text, NumberStyles numberStyles)
 
         if ((numberStyles & NumberStyles.AllowHexSpecifier) != 0)
         {
-            if (text.StartsWith("0X") || text.StartsWith("0x")) // support 0x hex while user input
+            if (text.StartsWith("0x") || text.StartsWith("0X")) // support 0x hex while user input
+            {
+                text = text.Substring(2);
+            }
+            else if (text.StartsWith("h'") || text.StartsWith("H'")) // support verilog hex while user input
             {
                 text = text.Substring(2);
             }
             else if (text.StartsWith("h") || text.StartsWith("H")) // support  hex while user input
             {
                 text = text.Substring(1);
             }
-            else if (text.StartsWith("h'") || text.StartsWith("H'")) // support  hex while user input
+        }
+        else
+#if NET8_0_OR_GREATER
+        if ((numberStyles & NumberStyles.AllowBinarySpecifier) != 0)
+#else

```

Comment:
This entire binary session should be in NETSTANDARD_2_1_OR_GREATER, because it's impossible to choose Binary in 2.0 condition, so there is no reason to cast to an invaldi interger.

Reply:
Yeah you are correct it should be 8.0.。。